src/benchmarks-julia.jl

```diff
@@ -1,62 +1,97 @@
-#!/usr/bin/env julia
+options = Base.JLOptions()
+
+@assert VERSION >= v"1.0.2"
+@assert options.can_inline == 1
+@assert options.check_bounds == 2
+@assert options.opt_level == 3
 
 using RigidBodyDynamics
-using StaticArrays
+using Random
+using LinearAlgebra
 
 include("models.jl")
 
-function benchmark_julia_rnea(model, log_filename)
-    robot = parse_urdf(Float64, PATH * "$model.urdf")
-    if model != "lwr"
-        for joint in out_joints(root_body(robot), robot)
-            floatingjoint = Joint(string(joint), frame_before(joint), frame_after(joint), QuaternionFloating{Float64}())
-            replace_joint!(robot, joint, floatingjoint)
-        end
+function write_times(log_filename, times)
+    file = open(log_filename, "w") do io
+        println.(Ref(io), times[2 : end]) # skip the first sample, which includes compilation time
     end
+    return nothing
+end
 
-    states = Array{MechanismState}(undef, NBT)
-    for i = 1:NBT
-        states[i] = MechanismState(robot)
-        rand!(states[i])
+function benchmark_julia_rnea(model, log_filename)
+    robot = parse_urdf(PATH * "$model.urdf", floating=model != "lwr")
+    state = MechanismState(robot)
+    result = DynamicsResult(robot)
+    qs = [similar(configuration(state)) for _ = 1 : NBT]
+    vs = [similar(velocity(state)) for _ = 1 : NBT]
+    v̇s = [similar(velocity(state)) for _ = 1 : NBT]
+    τ = similar(velocity(state))
+    times = Vector{Int}(undef, NBT)
+    Random.seed!(1)
+    for i = 1 : NBT
+        rand!(state)
+        qs[i] .= configuration(state)
+        vs[i] .= velocity(state)
+        rand!(v̇s[i])
     end
+    run_rnea_benchmark(state, qs, vs, v̇s, τ, result.jointwrenches, result.accelerations, times)
+    write_times(log_filename, times)
+    return nothing
+end
 
-    file = open(log_filename, "w")
-    for i = 1:NBT
+function run_rnea_benchmark(state, qs, vs, v̇s, τ, jointwrenches, accelerations, times)
+    # Do garbage collection of previously allocated objects now, rather than randomly
+    # in the middle of the benchmark.
+    # If garbage is generated during the benchmark run, so be it (it shouldn't in this case),
+    # but time spent doing garbage collection of objects allocated outside this loop shouldn't be included.
+    GC.gc()
+    for i in eachindex(qs)
         debut = time_ns()
-        inverse_dynamics(states[i], states[i].v)
+        copyto!(configuration(state), qs[i])
+        copyto!(velocity(state), vs[i])
+        setdirty!(state)
+        v̇ = v̇s[i]
+        inverse_dynamics!(τ, jointwrenches, accelerations, state, v̇)       
```

Comment:
Use the in-place version `inverse_dynamics!` instead of the convenience function `inverse_dynamics`. This version doesn't perform any dynamic memory allocation.